xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -418,7 +418,7 @@ public Result selectConfig(PickSubchannelArgs args) {
           }
           long select = random.nextLong(totalWeight);
           long accumulator = 0;
-          for (int i = 0; i < action.weightedClusters().size(); i++) {
+          for (int i = 0; ; i++) {
```

Comment:
To make it clear this is intentional, consider do/while.
Also might make sense to add a note that "Shouldn't happen" but we do want to crash if somehow it does happen.

Reply:
We should also say that we expect all weights to be positive numbers. We don't do any validation on weights, granted they are retrieved from uint proto. But negative numbers + zero won't result in an infinite loop, just an incorrect selection.
We'd get an infinite loop if all weights are zero, but in this case `random.nextLong` will throw before we even get there anyway.

Reply:
XdsRouteConfigureResource already verifies that the sum of weights is positive and not greater than uint32; added a comment. I also added code to parse the weight as unsigned when initially creating it.